File: problems/array/java/IsAnagram.java
import java.util.Arrays;

public class IsAnagram {
    
    public boolean anagram(char[] input1, char[] input2) {
        if (input1.length != input2.length) { return false; }
        
        Arrays.sort(input1);
        Arrays.sort(input2);

        for(int i=0; i<input1.length; i++) {
            if(input1[i] != input2[i]) {
                return false;
            }
        }
        
        return true;
    }

    public static void main(String[] args) {
        IsAnagram obj = new IsAnagram();
        System.out.println(obj.anagram(new char[]{'h', 'e', 'l', 'l', 'o'}, new char[]{'o', 'l', 'e', 'h', 'o'}));
        System.out.println(obj.anagram(new char[]{'r', 'e', 's', 't', 'f', 'u', 'l'}, new char[]{'f', 'l', 'u', 's', 't', 'e', 'r'}));
    }
}

File: problems/array/typescript/IsAnagram.ts
/*
    Construct an algorithm to check whether two words (or phrases) are anagrams or not!
    An anagram is a word or phrase formed by rearranging the letters of a different word or phrase,
    typically using all the original letters exactly once
*/

function isAnagram(input1: string, input2: string) : boolean {

    if (input1.length !== input2.length ) { return false; };
    const a1 = input1.split('').sort();
    const a2 = input2.split('').sort();

    for ( let i = 0; i < input1.length; i++ ) {
        if (a1[1] !== a2[i] ) {
            return false;
        }
    }

    return true;
}

